### PR TITLE
Fix + Refact (AllowFix)

### PR DESCRIPTION
**Pull request by ivandrofly**
/cc @niksedk

### CODE REVIEW

File: src/Forms/FixCommonErrors.cs [hunk 1]
```diff
@@ -67,7 +67,7 @@ public sealed partial class FixCommonErrors : Form
         private static readonly Regex UrlOrg = new Regex(@"\w\.org\b", RegexOptions.Compiled);
 
         private static readonly Regex ReAfterLowercaseLetter = new Regex(@"[a-zæøåäöéùáàìéóúñüéíóúñü]I", RegexOptions.Compiled);
-        private static readonly Regex ReBeforeLowercaseLetter = new Regex(@"I[a-zæøåäöéùàìéóúñüéíóúñü]", RegexOptions.Compiled);
+        private static readonly Regex ReBeforeLowercaseLetter = new Regex(@"I[a-zæøåäöéùáàìéóúñüéíóúñü]", RegexOptions.Compiled);
```
Comment (ivandrofly) [inline src/Forms/FixCommonErrors.cs]: <code>à</code> added

File: src/Forms/FixCommonErrors.cs [hunk 2]
```diff
@@ -2745,8 +2733,10 @@ public static string FixAloneLowercaseIToUppercaseLine(Regex re, string oldText,
                 s = s.Replace(">" + target + "</", ">I</");
             if (s.Contains(">" + target + " "))
                 s = s.Replace(">" + target + " ", ">I ");
-            if (s.Contains(">" + target + "" + Environment.NewLine))
```
Comment (ivandrofly) [inline src/Forms/FixCommonErrors.cs]: "" is equal to string.Empty

File: src/Forms/FixCommonErrors.cs [hunk 3]
```diff
@@ -2854,19 +2849,12 @@ private void Fix3PlusLines()
             for (int i = 0; i < Subtitle.Paragraphs.Count; i++)
             {
                 Paragraph p = Subtitle.Paragraphs[i];
-                string text = p.Text;
-
-                if (Utilities.CountTagInText(text, Environment.NewLine) > 1)
+                if (AllowFix(p, fixAction) && Utilities.CountTagInText(p.Text, Environment.NewLine) > 1)
```
Comment (ivandrofly) [inline src/Forms/FixCommonErrors.cs]: Check if AllowFix is <code>true</code> before doing any changes which may slow the performance

File: src/Forms/FixCommonErrors.cs (hunk 2 above)
Comment (niksedk) [inline src/Forms/FixCommonErrors.cs]: I like string.Empty better - not important but String.Empty is used in SE

File: src/Forms/FixCommonErrors.cs [hunk 4]
```diff
@@ -2207,27 +2207,27 @@ public static string FixStartWithUppercaseLetterAfterParagraph(Paragraph p, Para
 
             if (p.Text != null && p.Text.Contains(Environment.NewLine))
             {
-                string[] arr = p.Text.Replace("\r\n", "\n").Replace("\r", "\n").Split('\n');
+                var arr = p.Text.Split(Utilities.NewLineChars, StringSplitOptions.RemoveEmptyEntries);
```
Comment (niksedk) [inline src/Forms/FixCommonErrors.cs]: Not correct... we should probably make a extension method for string splitting...

Comment (ivandrofly) [inline src/Forms/FixCommonErrors.cs]: So should anything be changed in this patch?!

Comment (niksedk) [inline src/Forms/FixCommonErrors.cs]: Yes, this is buggy - the split does not work as it removes empty entries... a new extension method "SplitToLines" could be made so avoid duplicate code for line splitting

Comment (ivandrofly) [inline src/Forms/FixCommonErrors.cs]: Like:
<code>
        public static string[] SplitToLines(this string source)
        {
            return source.Replace("\r\n", "\n").Replace("\r", "\n").Split('\n');
        }
</code>

Comment (niksedk) [inline src/Forms/FixCommonErrors.cs]: :)

**Pull request closed without merging**